spec/artsy-eventstream/artsy/publisher_spec.rb

```diff
@@ -67,10 +68,33 @@
         routing_key: 'good.route',
         persistent: true,
         content_type: 'application/json',
+        headers: {},
         app_id: 'artsy'
       )
       Artsy::EventService::Publisher.publish_event(topic: 'test', event: event, routing_key: 'good.route')
     end
+    it 'passes through provided headers when calling publish on the exchange' do
+      conn = double
+      channel = double
+      exchange = double
+      allow(Artsy::EventService::RabbitMQConnection).to receive(:get_connection).with(no_args).and_return(conn)
+      allow(conn).to receive(:create_channel).and_return(channel)
+      allow(channel).to receive(:open?).and_return(true)
+      allow(channel).to receive(:topic).with('test', durable: true).and_return(exchange)
+      allow(channel).to receive(:confirm_select)
+      allow(channel).to receive(:wait_for_confirms).and_return(true)
+      allow(channel).to receive(:close)
+
+      expect(exchange).to receive(:publish).with(
+        JSON.generate(hello: true),
+        routing_key: 'good.route',
+        persistent: true,
+        content_type: 'application/json',
+        headers: {"x-foo": "bar"},
+        app_id: 'artsy'
+      )
+      Artsy::EventService::Publisher.publish_event(topic: 'test', event: event, routing_key: 'good.route', headers: {"x-foo": "bar"})
+    end
```

Comment:
:+1: this looks good - maybe some of those `allow` declarations could be `expect` so that you actually assert those calls.

Reply:
yes ☝️